### PR TITLE
services: Deprecate V1alpha

### PR DESCRIPTION
**Pull request by shivaspeaks**

### CODE REVIEW

**Review by ejona86 — APPROVED** [review]

**Comment by shivaspeaks** [comment]
The jobs are failing with warning, `Deprecated Gradle features were used in this build`.
Is it fine merge? @ejona86

**Comment by ejona86** [comment]
> The jobs are failing with warning, Deprecated Gradle features were used in this build.

That's not why. Javac is failing:

```
/home/runner/work/grpc-java/grpc-java/interop-testing/src/main/java/io/grpc/testing/integration/XdsTestServer.java:39: warning: [deprecation] ProtoReflectionService in io.grpc.protobuf.services has been deprecated
import io.grpc.protobuf.services.ProtoReflectionService;
                                ^
/home/runner/work/grpc-java/grpc-java/interop-testing/src/main/java/io/grpc/testing/integration/XdsTestClient.java:46: warning: [deprecation] ProtoReflectionService in io.grpc.protobuf.services has been deprecated
import io.grpc.protobuf.services.ProtoReflectionService;
                                ^
error: warnings found and -Werror specified

/home/runner/work/grpc-java/grpc-java/interop-testing/src/main/java/io/grpc/testing/integration/XdsTestServer.java:223: warning: [deprecation] ProtoReflectionService in io.grpc.protobuf.services has been deprecated
              .addService(ProtoReflectionService.newInstance())
                          ^
/home/runner/work/grpc-java/grpc-java/interop-testing/src/main/java/io/grpc/testing/integration/XdsTestServer.java:275: warning: [deprecation] ProtoReflectionService in io.grpc.protobuf.services has been deprecated
              .addService(ProtoReflectionService.newInstance())
                          ^
/home/runner/work/grpc-java/grpc-java/interop-testing/src/main/java/io/grpc/testing/integration/XdsTestClient.java:280: warning: [deprecation] ProtoReflectionService in io.grpc.protobuf.services has been deprecated
            .addService(ProtoReflectionService.newInstance())
                        ^
1 error
5 warnings

> Task :grpc-interop-testing:compileJava FAILED
```

> Is it fine merge?

No, because the test status is failing, and it isn't even flaky. (This would be true regardless of the error/warning.)

**Comment by shivaspeaks** [comment]
To resolve this, can I use @SuppressWarnings("deprecation") wherever this v1alpha is getting used?

**Comment by ejona86** [comment]
You can, but you can't put `@SuppressWarnings` on an import. So you need to either fully-qualify the usage or move the `@Deprecated` annotation to the methods, like `newInstance()` (still keeping the deprecation no the javadoc, probably).

File: interop-testing/src/main/java/io/grpc/testing/integration/XdsTestClient.java [hunk 1]
```diff
@@ -260,6 +260,7 @@ private static RpcType parseRpc(String rpc) {
     }
   }
 
+  @SuppressWarnings("deprecation")
```
Comment (ejona86) [inline interop-testing/src/main/java/io/grpc/testing/integration/XdsTestClient.java:263]: This suppressed deprecation for many things. We do want deprecation warnings. It is better to put the annotation on an assignment:
```java
@SuppressWarnings("deprecation")
BindableService oldReflectionService = ProtoReflectionService.newInstance();
```

Ditto for the other files.

File: services/src/test/java/io/grpc/protobuf/services/ProtoReflectionServiceTest.java [hunk 2]
```diff
@@ -71,7 +71,8 @@ public class ProtoReflectionServiceTest {
 
   private static final String TEST_HOST = "localhost";
   private MutableHandlerRegistry handlerRegistry = new MutableHandlerRegistry();
-  private BindableService reflectionService;
+  @SuppressWarnings("deprecation")
+  BindableService reflectionService = ProtoReflectionService.newInstance();
```
Comment (ejona86) [inline services/src/test/java/io/grpc/protobuf/services/ProtoReflectionServiceTest.java:75]: Why remove `private`?

**Review by ejona86 — APPROVED** [review]

**Review by kannanjgithub — APPROVED** [review]

**Pull request merged**